### PR TITLE
Adding `DataChain.column(...)` and fixing functions and types

### PR DESCRIPTION
**Pull request by ilongin**
This PR tries to fix general issue of `.mutate()` receiving expression or column without type, which ends up with error when .`save()` is called.

Couple of things are added / fixed:

- New `.column(name: str)` method in `DataChain` which returns `Column` instance with defined **type** by some column name from current schema of that `DataChain` instance. SQLAlchemy expressions that are build with columns with types have inferred type and those with columns without type ends up without inferred type (its `NullType` as result) so `.mutate()` should use this method instead raw `Column(...)` when adding expressions for new added columns.
- Redefining `avg` function with type `float`. Before, we were just re-importing this one from sqlalchemy and it didn't have specific type when used in expressions which would make whole expression without type. Now it will always have `float` type. Other functions that are re-imported directly from SQLAlchemy seems to have defined type in expressions and are working ok in `.mutate()` for example
- Fixing `sql_to_python()` function. Before, this was only checking for a subset of all possible SQL types and was failing if type was for example `INTEGER` which is DB specific and different from generic `Intteger`.
- Adding tests for `.mutate()` with complex expressions

### CODE REVIEW

**Review by dmpetrov — CHANGES_REQUESTED** [review]
Amazing change.

Comments are inline.

File: tests/unit/lib/test_datachain.py [hunk 1]
```diff
@@ -1258,3 +1262,56 @@ def test_show_limit():
...
+    assert c.name == "missing"
+    assert isinstance(c.type, NullType)
+
+
+def test_mutate_with_complex_expressions(catalog):
+    ds = DataChain.from_values(id=[1, 2, 3], name=["Jim", "Jon", "Joe"])
+
+    ds1 = ds.mutate(new=ds.column("id") - 1)
+    assert ds1.signals_schema.values["new"] is int
+    ds1.save("ds1")
```
[the remark this answers is not in the document]
Comment (ilongin) [reply tests/unit/lib/test_datachain.py]: Refactored tests (split them to multiple ones without save) and added one separate with save

File: src/datachain/lib/dc.py [hunk 2]
```diff
@@ -225,6 +226,16 @@ def schema(self) -> dict[str, DataType]:
         """Get schema of the chain."""
         return self._effective_signals_schema.values
 
+    def column(self, name: str) -> Column:
+        """Returns Column instance with a type if name is found in current schema,
+        otherwise returns plain Column without type.
+        """
+        for path, type_, _, _ in self.signals_schema.get_flat_tree():
+            if ".".join(path) == name:
```
[the remark this answers is not in the document]
Comment (ilongin) [reply src/datachain/lib/dc.py]: Applied

File: tests/unit/lib/test_datachain.py [hunk 3]
```diff
@@ -1258,3 +1261,78 @@ def test_show_limit():
...
+    ds = DataChain.from_values(id=[1, 2])
+    ds = ds.mutate(new=ds.column("id") / 2).save("mutated")
+
+    ds = DataChain(name="mutated")
+    assert ds.signals_schema.values["new"] is float
+    assert list(ds.collect("new")) == [0.5, 1.0]
+
+
+def test_mutate_with_expression_without_type(catalog):
+    with pytest.raises(CompileError):
```
Comment (dmpetrov) [inline tests/unit/lib/test_datachain.py:1337]: It should be wrapped with DC exception `class DataChainColumError(DataChainParamsError): ...`. In mutate() or signals_schema.mutate() methods, I guess.

Sqlalchemy is implementation detail and should not be exposed to users.

**Review by dmpetrov — APPROVED** [review]
Really nice implementation, thank you!

Now, we need to think if we can do a better job in case of not typed columns and do this conversion under the hood like:

```python
ds.mutate(col=(C("num") - 2) / C("count"))
--> 
ds.mutate(col=(ds.column("num") - 2) / ds.column("count"))
```

One comment is inline.

File: tests/unit/lib/test_datachain.py (hunk 3 above)
Comment (ilongin) [reply tests/unit/lib/test_datachain.py]: Done

**Comment by ilongin** [comment]
> Really nice implementation, thank you! One comment is inline.
> 
> Next steps in order to fully close #148 (not in this PR): We need to think if we can do a better job in case of not typed columns and do this conversion under the hood like:
> 
> ```python
> ds.mutate(col=(C("num") - 2) / C("count"))
> --> 
> ds.mutate(col=(ds.column("num") - 2) / ds.column("count"))
> ```

Yes, this is possible to do. We need to traverse expression. Problematic part could be to traverse function arguments as I'm not sure they offer hat in API but I guess it should be possible

**Review by dreadatour — APPROVED** [review]
❤️

**Pull request merged**